backend/python/app/services/implementations/story_service.py

```diff
@@ -152,23 +154,49 @@ def get_story_translations(
                 .all()
             )
 
-            story_translations = (
-                StoryTranslation.query.join(
-                    Story, Story.id == StoryTranslation.story_id
+            translator = aliased(User)
+            reviewer = aliased(User)
+
+            story_translation_data = (
+                db.session.query(StoryTranslation, translator, reviewer)
+                .join(Story, Story.id == StoryTranslation.story_id)
+                .join(
+                    translator,
+                    StoryTranslation.translator_id == translator.id,
+                    isouter=True,
+                )
+                .join(
+                    reviewer, StoryTranslation.reviewer_id == reviewer.id, isouter=True
                 )
                 .filter(*filters)
                 .order_by(Story.id)
                 .all()
             )
 
-            for i in range(len(story_translations)):
+            story_translations = []
+
+            for i in range(len(story_translation_data)):
+                story_translation, translator, reviewer = story_translation_data[i]
                 story = next(
```

Comment:
using `next` is pretty massive brained :-0 

I'm not very familiar with next but it seems like it needs to be used with a generator for max efficiency, does the `stories` satisfy that? https://stackoverflow.com/questions/31040017/is-next-in-python-really-that-fast
Either way, still happy to keep using next :)

Reply:
oh hmm since we are always expecting just one result from the filtering, i used `next()` instead of converting first to list and returning the item at 0 index:
`story = list(filter(lambda x: x.id == story_translation.story_id, stories))[0]` <- do you think this would be clearer?

inspiration from:
https://stackoverflow.com/questions/34026045/return-the-first-element-or-none-from-filter
https://docs.python.org/3/library/functions.html#next

Reply:
`story = list(filter(lambda x: x.id == story_translation.story_id, stories))[0]` isn't particularly more clear imo

Happy with the code as is!